### DISCUSSION TITLE
Installation issue: error in pyvista setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'

### DISCUSSION THREAD

**Discussion by tanki1001** [pyvista/pyvista#5758] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
#None of the following way to install pyvista works. Neither in the terminal nor in a .ipynb file
!pip install pyvista
!pip install 'pyvista' jupyterlab
!pip install pyvista[all]
!pip install 'pyvista[all]' jupyterlab
```

### Description

I try to install pyvista in a jupyter notebook, with pip and it returns:

  Using cached pyvista-0.25.1.tar.gz (1.2 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in pyvista setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Note: you may need to restart the kernel to use updated packages.

### System Information

```shell
#while it's an installation problem, this is not very surprising:

-------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[4], line 2
      1 # Get system info
----> 2 import pyvista as pv
      3 print(pv.Report())

ModuleNotFoundError: No module named 'pyvista'
```

**Comment by user27182** [comment]
Try `!pip install --upgrade pyvista` or `!pip install pyvista==0.43`.

Using `pip install` without specifying a version will result in pip trying to use a cached version of the package if available. In your case, it's trying to use a cached version of `pyvista v0.25`. This is an old version of pyvista which is no longer supported.

**Reply by tanki1001** [reply]
Thank's a lot for your quick response!

I'm afraid none of the command line works. However the second one provides me a different error:

Collecting pyvista==0.43
  Using cached pyvista-0.43.0-py3-none-any.whl.metadata (14 kB)
Requirement already satisfied: matplotlib>=3.0.1 in /usr/local/lib/python3.10/dist-packages (from pyvista==0.43) (3.8.0)
Requirement already satisfied: numpy>=1.21.0 in /usr/local/lib/python3.10/dist-packages (from pyvista==0.43) (1.23.2)
Requirement already satisfied: pillow in /usr/local/lib/python3.10/dist-packages (from pyvista==0.43) (10.0.1)
Collecting pooch (from pyvista==0.43)
  Using cached pooch-1.8.1-py3-none-any.whl.metadata (9.5 kB)
Collecting scooby>=0.5.1 (from pyvista==0.43)
  Using cached scooby-0.9.2-py3-none-any.whl.metadata (15 kB)
INFO: pip is looking at multiple versions of pyvista to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement vtk (from pyvista) (from versions: none)
ERROR: No matching distribution found for vtk


Indeed, when I run pip install vtk I have the two last lines, without the (from pyvista) for sure.
I don't know if it's relevant but I'm actually using a FEniCSx docker, and the command `pip install pyvista` works in my terminal on my machine in local, so I'll also open a ticket on FEniCSx community.

Thank's again!

**Reply by user27182** [reply]
This seems like an issue with your environment. It could be that newer versions of vtk are not supported on your system and/or incompatible with other packages currently installed in your environment. If you try `pip install --upgrade vtk` then `pip freeze | grep vtk`, what version of vtk is installed?

You can also try installing pyvista with anaconda, e.g. `conda install -c conda-forge pyvista`. See https://docs.pyvista.org/version/stable/getting-started/

Ideally, it is best to install pyvista in a new environment. Anaconda can be used for this, or python's `venv` module works too.

**Reply by tanki1001** [reply]
Im not sure what `pip freeze | grep vtk` is for, but when I use is after `pip install --upgrade vtk` (That returns me the same original error: ERROR: Could not find a version that satisfies the requirement vtk (from versions: none)), I have nothing as output. And rerunning the installing gives me the same error.
But yes, something is going on with the docker environnement...
